File: clients/python/examples/basic.py
"""
MilanSQL Python Client — Basic TCP example

Requires: MilanSQL server running on port 4406
  .\\build\\milansql.exe --server --port 4406
"""

import milansql

with milansql.connect(host='localhost', port=4406) as conn:
    cur = conn.cursor()

    # Create table
    cur.execute(
        "CREATE TABLE IF NOT EXISTS produkte "
        "(id INT PRIMARY KEY AUTO_INCREMENT, name TEXT, preis INT)"
    )
    print("Tabelle erstellt.")

    # Insert rows with parameterized queries
    cur.execute("INSERT INTO produkte VALUES (NULL, %s, %s)", ("Laptop", 1200))
    cur.execute("INSERT INTO produkte VALUES (NULL, %s, %s)", ("Maus", 25))
    cur.execute("INSERT INTO produkte VALUES (NULL, %s, %s)", ("Monitor", 350))
    print("3 Zeilen eingefügt.")

    # SELECT with filter
    cur.execute("SELECT * FROM produkte WHERE preis > %s ORDER BY preis DESC", (100,))
    print(f"\nProdukte teurer als 100 ({cur.rowcount} Ergebnisse):")
    print(f"  Spalten: {[col[0] for col in cur.description]}")
    for row in cur.fetchall():
        print(f"  {row}")

    # Aggregate
    cur.execute("SELECT COUNT(*), AVG(preis), MAX(preis) FROM produkte")
    row = cur.fetchone()
    print(f"\nStatistik: Anzahl={row[0]}, Durchschnitt={row[1]}, Maximum={row[2]}")

File: clients/nodejs/examples/basic.js
'use strict';
/**
 * MilanSQL Node.js Client — Basic example
 *
 * Requires: MilanSQL HTTP server on port 8080
 *   .\build\milansql.exe --http --port 8080
 */

const milansql = require('../index');

async function main() {
  const conn = milansql.connect({ host: 'localhost', port: 8080 });

  // Server status
  const status = await conn.status();
  console.log('Server Status:', status);

  // Create table
  await conn.query(
    'CREATE TABLE IF NOT EXISTS users ' +
    '(id INT PRIMARY KEY AUTO_INCREMENT, name TEXT, score INT)'
  );
  console.log('\nTabelle erstellt.');

  // Insert rows
  for (const [name, score] of [['Alice', 95], ['Bob', 82], ['Charlie', 70]]) {
    const r = await conn.query(`INSERT INTO users VALUES (NULL, ${name}, ${score})`);
    console.log(`  INSERT: ${r.message || 'OK'}`);
  }

  // SELECT via POST
  const result = await conn.query('SELECT * FROM users ORDER BY score DESC');
  console.log(`\nAlle Benutzer (${result.rowCount} Zeilen):`);
  console.log('  Spalten:', result.columns);
  for (const row of result.rows) {
    console.log(' ', row);
  }

  // SELECT via GET
  const filtered = await conn.queryGet('SELECT name, score FROM users WHERE score > 80');
  console.log('\nBenutzer mit Score > 80:');
  for (const row of filtered.rows) {
    console.log(' ', row);
  }

  // Metadata
  const tables = await conn.tables();
  console.log('\nTabellen:', tables);

  const schemas = await conn.schemas();
  console.log('Schemas:', schemas);

  conn.close();
}

main().catch(console.error);
